cli/src/infragpt/llm_adapter.py

```
@@ -1,332 +1,223 @@
+                    # Handle interrupt during tool execution
+                    console.print("\n[yellow]Tool execution cancelled by user.[/yellow]")
+                    return
+                
+        except ToolExecutionCancelled:
+            # User cancelled - propagate without wrapping
+            raise
+        except Exception as e:
+            error_msg = f"Streaming failed: {e}"
+            if self.verbose:
+                console.print(f"[red]{error_msg}[/red]")
+            raise LLMError(error_msg)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)